src/extension/features/budget/cover-overspending-from-future/index.js

```diff
@@ -0,0 +1,163 @@
+import { Feature } from 'toolkit/extension/features/feature';
+import { /* getSelectedMonth, */ getEntityManager, isCurrentRouteBudgetPage } from 'toolkit/extension/utils/ynab';
+import { formatCurrency } from 'toolkit/extension/utils/currency';
+import { l10n } from 'toolkit/extension/utils/toolkit';
+import { controllerLookup } from 'toolkit/extension/utils/ember';
+
+const BUTTON_CLASS = 'tk-button-cover-from-future';
+
+export class CoverOverspendingFromFuture extends Feature {
+  injectCSS() {
+    return require('./index.css');
+  }
+
+  shouldInvoke() {
+    return isCurrentRouteBudgetPage();
+  }
+
+  invoke() {
+    this.verifyButtonDisplay();
+  }
+
+  observe(changedNodes) {
+    if (!this.shouldInvoke()) return;
+    if (changedNodes.has('budget-table-row is-sub-category is-checked')) {
+      this.invoke();
+    }
+  }
+
+  verifyButtonDisplay() {
+    const checkedRows = controllerLookup('budget').get('checkedRows');
+
+    // Do nothing with the button if more than one row is selected. (maybe change in the future)
+    if (checkedRows.length !== 1) {
+      return this.removeButton();
+    }
+
+    const category = checkedRows[0];
+    const { isMasterCategory, available, budgeted } = category.getProperties('isMasterCategory', 'available', 'budgeted');
+
+    // TODO: Test
+    console.log(budgeted);
+
+    // We can't budget for master categories
+    // We don't need this if available is not negative
+    if (isMasterCategory || available >= 0) {
+      return this.removeButton();
+    }
+
+    // Search for the button
+    let button = $(`.${BUTTON_CLASS}`);
+
+    // Create button if it doesn't exist already
+    if (!button.length) {
+      button = $('<button>', {
+        class: `budget-inspector-button ${BUTTON_CLASS}`
+      }).click(this.coverFromFuture.bind(this, category));
+
+      // Add the button at the top.
+      $('.inspector-quick-budget button').first().before(button);
+    }
+
+    const coverAmount = Math.abs(available);
+    const fCoverAmount = '+' + formatCurrency(coverAmount);
+
+    // Update content
+    button
+      .empty()
+      .data('coverAmount', coverAmount)
+      .append(l10n('toolkit.budget.coff.coverFromFuture', 'Cover From Future'))
+      .append(': ')
+      .append($('<strong>', { class: 'user-data', title: fCoverAmount })
+        .append($('<span>', { class: 'user-data currency positive' })
+          .text(fCoverAmount)));
+
+    return true;
+  }
+
+  coverFromFuture(category) {
+    // TODO
+    console.log('Yeah, cover from future clicked!');
+    console.log(category);
+    const categories = this.getRowsForCategory(category);
+    console.log(categories);
+  }
+
+  removeButton() {
+    const button = $(`.${BUTTON_CLASS}`);
```

Comment:
you can actually just do `$(`.${BUTTON_CLASS}`).remove();` and don't need to return anything (I don't see you using the return value atm anyway).

jquery gives you a no-op jquery object if it doesn't find the selector so you don't need all the other logic